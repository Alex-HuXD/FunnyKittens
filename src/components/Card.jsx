import React from "react";

function Card({ name, email, id }) {
  return (
    <div className=" bg-light-green white-90 dib br3 pa2 ma3 shadow-5 grow">
      <img
        src={`https://robohash.org/${id}?set=set4&size=250x250`}
        alt="card"
      />
      <div>
        <p>{name}</p>
        <p>{email}</p>
      </div>
    </div>
  );
}

export default Card;
