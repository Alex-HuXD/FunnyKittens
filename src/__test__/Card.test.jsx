import { render, screen } from '@testing-library/react';

import Card from '../components/Card';

const mockedProps = {
  name: 'testName',
  email: 'testEmail',
  id: 1,
};
describe('Card', () => {
  it('should render name, email and id', () => {
    render(<Card {...mockedProps} />);

    const nameElement = screen.getByText(/testname/i);
    const emailElement = screen.getByText(/testemail/i);
    const imgElement = screen.getByRole('img');

    expect(nameElement).toBeInTheDocument();
    expect(emailElement).toBeInTheDocument();
    expect(imgElement).toBeInTheDocument();
  });

  it('should display same as snapshot && if style change is intentional then snapshot must be updated', () => {
    render(<Card {...mockedProps} />);
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
