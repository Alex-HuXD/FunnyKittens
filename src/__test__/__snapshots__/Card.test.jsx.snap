// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should display same as snapshot && if style change is intentional then snapshot must be updated 1`] = `"<div><div class=\\" bg-light-green white-90 dib br3 pa2 ma3 shadow-5 grow\\"><img src=\\"https://robohash.org/1?set=set4&amp;size=250x250\\" alt=\\"card\\"><div><p>testName</p><p>testEmail</p></div></div></div>"`;
